pkg/network/tcp_peer.go

```
@@ -267,18 +269,30 @@ func (p *TCPPeer) StartProtocol() {
 func (p *TCPPeer) Handshaked() bool {
 	p.lock.RLock()
 	defer p.lock.RUnlock()
+	return p.handshaked()
+}
+
+// handshaked is internal unlocked version of Handshaked().
+func (p *TCPPeer) handshaked() bool {
 	return p.handShake == (verAckReceived | verAckSent | versionReceived | versionSent)
 }
 
+// IsFullNode returns whether the node has full capability or TCP/WS only.
+func (p *TCPPeer) IsFullNode() bool {
+	p.lock.RLock()
+	defer p.lock.RUnlock()
+	return p.handshaked() && p.isFullNode
+}
+
 // SendVersion checks for the handshake state and sends a message to the peer.
 func (p *TCPPeer) SendVersion() error {
-	msg := p.server.getVersionMsg()
+	msg, err := p.server.getVersionMsg()
```

This error is not handled.